frontend/src/metabase/parameters/components/ValuesSourceModal/ValuesSourceTypeModal.tsx

```diff
@@ -383,6 +387,10 @@ const ListSourceModal = ({
     [onChangeSourceConfig],
   );
 
+  const hasCustomLabels = sourceConfig.values?.some(
+    value => Array.isArray(value) && value.length === 2,
```

Comment:
We could really use named helpers that deal with the indexes and array lengths :pray: